actioncable/test/client_test.rb

```diff
@@ -98,7 +105,26 @@ def with_puma_server(rack_app = ActionCable.server, port = 3099)
       end
     end
   end
+  if defined?(::Iodine)
+    def with_iodine_server(rack_app = ActionCable.server, port = 3099, block)
+      ::Iodine.listen2http(app: rack_app, port: port.to_s, address: "127.0.0.1")
+      ::Iodine.workers = 1 # don't cluster the test
+      ::Iodine.threads = 1 # one for the server another for the task
+      t = Thread.new { ::Iodine.start }
+      block.call(port)
+      ::Iodine.stop
+      t.join
```

Comment:
Sure, I'll do that, though it's quite redundant for iodine.

Iodine protects against exceptions, so it's only the code block that could raise an exception (the main thread).

I wasn't not sure if I should have protected against test related exceptions, but now I know I should.

Reply:
Using the `ensure` class also communicates intent: that the server should be stopped, regardless of whether the block raises. Otherwise, it's unclear whether this possibility was missed, or that the block is unconditionally responsible for exception handling, despite the Puma case indicating otherwise.